src/app/ApplicationTemplate.Shared/Framework/DependencyInjection/RegistrationServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+	public static class RegistrationServiceCollectionExtensions
+	{
+		/// <summary>
+		/// Registers all services in <paramref name="assembly"/> that are marked with <see cref="RegisterServiceAttribute"/>.
+		/// </summary>
+		/// <param name="services">The <see cref="IServiceCollection"/>.</param>
+		/// <param name="assembly">The <see cref="Assembly"/> to scan.</param>
+		/// <returns>The <see cref="IServiceCollection"/> with services registered.</returns>
+		public static IServiceCollection AutoRegisterServicesFromAssembly(this IServiceCollection services, Assembly assembly = null)
+		{
+			assembly = assembly ?? Assembly.GetCallingAssembly();
+			services.Scan(scan =>
```

Comment:
What's the impact on startup performance?

Reply:
Not measured exactly, as I have a lack of context about your own performance benchmark. But so far, the perceived startup time is pretty much the same. The scanned assemblies are not that big in our contexts.

Reply:
You can compare by going to the diagnostics screen (by tapping the version number 3 times) and looking at the "Startup time" section. The "Initialize" subsection is where the services are registered.